p2p/pubsubManager/gossipsub.go

```diff
@@ -43,22 +45,36 @@ func NewGossipSubManager(ctx context.Context, h host.Host) (*PubsubManager, erro
 		make(map[string]*pubsub.Subscription),
 		make(map[string]*pubsub.Topic),
 		nil,
+		utils.MakeGenesis().ToHeader().Hash(),
 		nil,
 	}, nil
 }
 
 func (g *PubsubManager) SetQuaiBackend(consensus quai.ConsensusAPI) {
-	g.consensus = consensus
+	g.UnsubscribeAll()      // First unsubscribe from existing topics, if already registered
```

Comment:
Curious why this is needed if the quaibackend is supposed to be set once at the beginning.

Reply:
Reentrancy. There is nothing preventing the backend from being set multiple times. If you set a new backend, your old subscriptions are no longer valid.

I agree, we probably never will reset the backend (but who knows what some future failure mode logic requires). As long as we have a method which supports setting a new back end, we need subscriptions to be safely managed (and thus reset) when this occurs.

_iff_ anyone does set a new back end, they can simply re-subscribe to topics of interest.